billing/lib/ucan-stream.js

```diff
@@ -19,9 +19,9 @@ export const findSpaceUsageDeltas = messages => {
     let resource
     /** @type {number|undefined} */
     let size
-    if (isReceiptForCapability(message, ServiceBlobCaps.allocate) && isServiceBlobAllocateSuccess(message.out)) {
+    if (isReceiptForCapability(message, ServiceBlobCaps.accept) && isServiceBlobAcceptSuccess(message.out)) {
       resource = message.value.att[0].nb?.space
-      size = message.out.ok.size
+      size = message.value.att[0].nb?.blob.size
```

Comment:
Ah, and this works because `value` points to the original _invocation_ for blob accept, which _does_ include the size.

Reply:
The only problem with this approach is that it doesn't take into account whether the blob is stored in the space already or not. The receipt for `blob/allocate`will have `size: 0` when the blob already exists in the space.

It's probably better than the current situation though...

Reply:
If `size: 0`, we are not adding it to the diff table.

Reply:
@hannahhoward WDYT shall we merge and release this? IMHO it's better than the status quo but still needs more work.

Reply:
I understand now. I confirmed this in the code, and it's worse than you thought. We're actually inserting duplicate values of `bafyxyz` every time we store it. This happens because in this flow, the `blob/accept` is executed, resulting in two receipts for it.

Reply:
If the blob already exists, we return success for `blob/allocate` with a size of 0 and `out.ok.address` set to `null`. However, if this occurs, we immediately return the receipt for `http/put`, and if it has succeeded, we execute `blob/accept`, leading to two receipts in the stream.

Reply:
I think in that case we should just not generate those receipts early.

Reply:
Would love to catch up on this one - if it doesn't come up at sprint planning or we don't have enough time to get into it I'd love to sync briefly after that if y'all are free!

Reply:
OK - in our chat this morning we decided to ship this change since it will significantly decrease the impact of the duplicate diff issue, and to deprecate the ucan stream entirely as part of the upload-service work - @alanshaw should we create one more item under that upload service epic for this work? I'm happy to go file that and fill in some basic details if so...

Reply:
I summarized the plan moving forward here: https://github.com/storacha/RFC/pull/39